plugins/nexus-repository-conda/src/main/java/org/sonatype/nexus/repository/conda/internal/hosted/CondaHostedFacetImpl.java

```diff
@@ -0,0 +1,262 @@
+            upload(entry.getKey() + "/repodata.json", content);
+        }
+    }
+
+    private RepoData takeOrCreate(Map<String, RepoData> architectures, String group) {
+        if (!architectures.containsKey(group)) {
+            architectures.put(group, createRepoData(group));
+        }
+        return architectures.get(group);
+    }
+
+    private RepoData createRepoData(String arch) {
+        RepoData repoData = new RepoData();
+        repoData.getInfo().setSubdir(arch);
+        return repoData;
+    }
+
+    @TransactionalStoreMetadata
+    private Asset getOrCreateAsset(final CondaPath condaPath, final Repository repository) {
+        final String componentName = condaPath.getCoordinates()
+                .map(coordinates -> coordinates.getPackageName())
+                .orElse(condaPath.getFileName());
+
+        final String componentGroup = getGroup(condaPath.getPath());
+        final StorageTx tx = UnitOfWork.currentTx();
+        final Bucket bucket = tx.findBucket(getRepository());
+        String path = condaPath.getPath();
+
+        log.debug("Find component with name " + componentName + " and path: " + path);
+        Asset asset = findComponent(tx, repository, condaPath)
+                .map(component -> {
+                            // UPDATE
+                            log.debug("Component exists: " + component.group() + " - " + component.name());
+                            log.debug("Find asset " + path);
+                            return Optional
+                                    .ofNullable(tx.findAssetWithProperty(P_NAME, path, component))
+                                    .orElseGet(() -> {
+                                        log.info("Asset doesn't exist.  Create it");
+                                        return tx.createAsset(bucket, component);
+                                    });
+                        }
+                )
+                .orElseGet(() -> {
+                    // CREATE
+                    log.debug("Create new component and asset");
+                    Component component = condaPath.getCoordinates()
+                            .map(coordinates -> tx.createComponent(bucket, getRepository().getFormat())
+                                    .group(componentGroup)
+                                    .name(componentName)
+                                    .version(coordinates.getVersion())
+                            ).orElse(tx.createComponent(bucket, getRepository().getFormat())
+                                    .group(componentGroup)
+                                    .name(componentName));
+                    tx.saveComponent(component);
+                    return tx.createAsset(bucket, component);
+                });
+
+        asset.name(condaPath.getPath());
+        asset.markAsDownloaded();
+        return asset;
+    }
+
+
+    private static Optional<Component> findComponent(final StorageTx tx,
```

Comment:
looks like you have same functionality in CondaDataAccess, maybe leave one implementation?

Reply:
I have run out of time to spend on this code. As mention above there are a lot of changes which should be introduced but this can be done as follow-up change.

Reply:
Well, in my opinion it's better to introduce this changes now. I propose to left methods, witch work with some repos things like assets, components in DataAccess class or create separate facet for this activities and use it for hosted or proxy facet. I did something same in HELM format, maybe it will be helpful for you: https://github.com/sonatype-nexus-community/nexus-repository-helm/pull/86
What do you think about it?

Reply:
I will be happy to work on it and fix some part of the code but I need to know the nexus team is willing to accept my changes. I have been trying to submit some code from three months so far and the feedback is really limited.

Till I confirm this will give us a solid solution for our needs I will not get more time for this codebase...